In [426]:
import pandas as pd
import plotly.express as px
from scipy.stats import ttest_ind

In [427]:
qwen_results_og = pd.read_csv('results/averaged_flash-holmes.csv')
qwen_results_og_halfp = pd.read_csv('results/averaged_flash-holmes_halfp.csv')
qwen_results_og_8bit = pd.read_csv('results/averaged_flash-holmes_8bit.csv')
qwen_results_og_4bit = pd.read_csv('results/averaged_flash-holmes_4bit.csv')
other_models_og = pd.read_csv('data/leaderboards/flash-holmes.csv')

In [428]:
other_models = other_models_og[other_models_og['probe type'].isin(['sentence pair classification', 'sentence classification'])]
qwen_results = qwen_results_og[qwen_results_og['probing_dataset'].isin(other_models['probing dataset'].unique())].reset_index(drop=True)
qwen_results_halfp =  qwen_results_og_halfp[qwen_results_og_halfp['probing_dataset'].isin(other_models['probing dataset'].unique())].reset_index(drop=True)
qwen_results_8bit =  qwen_results_og_8bit[qwen_results_og_8bit['probing_dataset'].isin(other_models['probing dataset'].unique())].reset_index(drop=True)
qwen_results_4bit =  qwen_results_og_4bit[qwen_results_og_4bit['probing_dataset'].isin(other_models['probing dataset'].unique())].reset_index(drop=True)

In [429]:
other_models

,Unnamed: 0,probing dataset,linguistic subfield,linguistic phenomena,probe type,train portion,probe,EleutherAI/pythia-1b,EleutherAI/pythia-1b-deduped,EleutherAI/pythia-1.4b,...,mistralai/Mistral-7B-v0.1,mistralai/Mixtral-8x7B-Instruct-v0.1,mistralai/Mixtral-8x7B-v0.1,databricks/dolly-v2-12b,WizardLM/WizardLM-13B-V1.2,ibm/merlinite-7b,ibm/labradorite-13b,allenai/tk-instruct-11b-def,sentence-transformers/average_word_embeddings_glove.840B.300d,sentence-transformers/average_word_embeddings_glove.6B.300d
0,0,SemAntoNeg,reasoning,negation,sentence pair classification,0.001953,SemAntoNeg,0.418881,0.408370,0.435316,...,0.444773,0.431906,0.397140,0.436657,0.422245,0.456564,0.419699,0.478874,0.501787,0.508634
6,7,bioscope-org-negation,reasoning,negation,sentence classification,0.001953,bioscope-org-negation,0.522602,0.535114,0.496731,...,0.563916,0.428151,0.418483,0.536987,0.536703,0.504667,0.583060,0.468011,0.488949,0.489449
7,8,blimp-adjunct_island,syntax,island effects,sentence classification,0.001953,blimp-adjunct_island,0.487530,0.450742,0.488316,...,0.474936,0.409880,0.359386,0.534357,0.411795,0.450534,0.470166,0.815800,0.454830,0.483301
8,9,blimp-anaphor_gender_agreement,morphology,anaphor agreement,sentence classification,0.001953,blimp-anaphor_gender_agreement,0.494194,0.419702,0.436624,...,0.443844,0.463560,0.376095,0.498254,0.476573,0.470043,0.485628,0.864270,0.563984,0.490832
9,10,blimp-anaphor_number_agreement,morphology,anaphor agreement,sentence classification,0.001953,blimp-anaphor_number_agreement,0.452827,0.437381,0.477801,...,0.414584,0.435005,0.375546,0.550244,0.479444,0.446371,0.627451,0.836923,0.494042,0.482688
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1674,196,zorro-local_attractor-in_question_with_aux,syntax,local attractor,sentence classification,0.500000,zorro-local_attractor-in_question_with_aux,0.513786,0.479564,0.468112,...,0.414062,0.462480,0.334440,0.358142,0.887876,0.466828,0.370462,0.941996,0.892224,0.871572
1675,197,zorro-npi_licensing-matrix_question,syntax,negative polarity item licensing,sentence classification,0.500000,zorro-npi_licensing-matrix_question,0.999250,1.000000,0.993500,...,0.999000,1.000000,1.000000,0.994499,0.902727,0.999500,1.000000,0.993500,0.499352,0.497626
1676,198,zorro-npi_licensing-only_npi_licensor,syntax,negative polarity item licensing,sentence classification,0.500000,zorro-npi_licensing-only_npi_licensor,0.999375,0.996000,0.998000,...,1.000000,1.000000,1.000000,1.000000,0.600561,1.000000,0.999500,0.982484,0.493452,0.483635
1677,199,zorro-quantifiers-existential_there,semantics,quantifiers,sentence classification,0.500000,zorro-quantifiers-existential_there,0.999750,0.999750,0.999250,...,0.992498,0.995248,0.999500,0.999500,0.979491,0.999250,1.000000,0.971477,0.499184,0.498885


In [430]:
qwen_results

,Unnamed: 0,probing_dataset,model_name,score
0,0,SemAntoNeg,Qwen__Qwen2.5-0.5B,0.002807
1,1,SemAntoNeg,Qwen__Qwen2.5-0.5B-Instruct,0.004167
2,2,SemAntoNeg,Qwen__Qwen2.5-0.5B-Instruct-AWQ,0.000000
3,3,SemAntoNeg,Qwen__Qwen2.5-1.5B,0.000000
4,4,SemAntoNeg,Qwen__Qwen2.5-1.5B-Instruct,0.000000
...,...,...,...,...
1543,1950,zorro-quantifiers-superlative,Qwen__Qwen2.5-3B-Instruct,0.967180
1544,1951,zorro-quantifiers-superlative,Qwen__Qwen2.5-3B-Instruct-AWQ,0.944393
1545,1952,zorro-quantifiers-superlative,Qwen__Qwen2.5-7B,0.528964
1546,1953,zorro-quantifiers-superlative,Qwen__Qwen2.5-7B-Instruct,0.722753


In [431]:
qwen_results_og_halfp

,Unnamed: 0,probing_dataset,model_name,score
0,0,SemAntoNeg,Qwen__Qwen2.5-0.5B-Instruct,0.000000
1,1,SemAntoNeg,Qwen__Qwen2.5-0.5B-Instruct-AWQ,0.000000
2,2,SemAntoNeg,Qwen__Qwen2.5-1.5B-Instruct,0.000000
3,3,SemAntoNeg,Qwen__Qwen2.5-1.5B-Instruct-AWQ,0.000000
4,4,SemAntoNeg,Qwen__Qwen2.5-3B-Instruct,0.000000
...,...,...,...,...
1024,1024,zorro-quantifiers-superlative,Qwen__Qwen2.5-1.5B-Instruct,0.948039
1025,1025,zorro-quantifiers-superlative,Qwen__Qwen2.5-1.5B-Instruct-AWQ,0.928130
1026,1026,zorro-quantifiers-superlative,Qwen__Qwen2.5-3B-Instruct,0.968133
1027,1027,zorro-quantifiers-superlative,Qwen__Qwen2.5-3B-Instruct-AWQ,0.938919


In [432]:
 qwen_results_8bit

,Unnamed: 0,probing_dataset,model_name,score
0,0,SemAntoNeg,Qwen__Qwen2.5-0.5B-Instruct,0.000000
1,1,SemAntoNeg,Qwen__Qwen2.5-1.5B-Instruct,0.000000
2,2,SemAntoNeg,Qwen__Qwen2.5-3B-Instruct,0.000000
3,3,SemAntoNeg,Qwen__Qwen2.5-7B-Instruct,0.000000
4,8,bioscope-org-negation,Qwen__Qwen2.5-0.5B-Instruct,0.348465
...,...,...,...,...
511,583,zorro-quantifiers-existential_there,Qwen__Qwen2.5-7B-Instruct,0.837179
512,584,zorro-quantifiers-superlative,Qwen__Qwen2.5-0.5B-Instruct,0.766732
513,585,zorro-quantifiers-superlative,Qwen__Qwen2.5-1.5B-Instruct,0.966251
514,586,zorro-quantifiers-superlative,Qwen__Qwen2.5-3B-Instruct,0.934165


In [433]:
qwen_results_4bit

,Unnamed: 0,probing_dataset,model_name,score
0,0,SemAntoNeg,Qwen__Qwen2.5-0.5B-Instruct,0.000000
1,1,SemAntoNeg,Qwen__Qwen2.5-1.5B-Instruct,0.000000
2,2,SemAntoNeg,Qwen__Qwen2.5-3B-Instruct,0.000000
3,3,SemAntoNeg,Qwen__Qwen2.5-7B-Instruct,0.000000
4,8,bioscope-org-negation,Qwen__Qwen2.5-0.5B-Instruct,0.354660
...,...,...,...,...
511,583,zorro-quantifiers-existential_there,Qwen__Qwen2.5-7B-Instruct,0.851654
512,584,zorro-quantifiers-superlative,Qwen__Qwen2.5-0.5B-Instruct,0.786813
513,585,zorro-quantifiers-superlative,Qwen__Qwen2.5-1.5B-Instruct,0.967719
514,586,zorro-quantifiers-superlative,Qwen__Qwen2.5-3B-Instruct,0.791649


In [434]:
mapping_subfield = other_models_og[['probing dataset', 'linguistic subfield']].drop_duplicates()
mapping_phenomena = other_models_og[['probing dataset', 'linguistic phenomena']].drop_duplicates()

In [435]:
qwen_results = qwen_results.merge(
    mapping_subfield,
    how='left',
    left_on='probing_dataset',
    right_on='probing dataset'
).merge(
    mapping_phenomena,
    how='left',
    left_on='probing_dataset',
    right_on='probing dataset'
).rename(columns={'probing_dataset': 'probe'}).drop(columns=['Unnamed: 0', 'probing dataset_x', 'probing dataset_y'])
qwen_results['model_name'] = qwen_results['model_name'].str.replace('__', '/')
qwen_results

,probe,model_name,score,linguistic subfield,linguistic phenomena
0,SemAntoNeg,Qwen/Qwen2.5-0.5B,0.002807,reasoning,negation
1,SemAntoNeg,Qwen/Qwen2.5-0.5B-Instruct,0.004167,reasoning,negation
2,SemAntoNeg,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.000000,reasoning,negation
3,SemAntoNeg,Qwen/Qwen2.5-1.5B,0.000000,reasoning,negation
4,SemAntoNeg,Qwen/Qwen2.5-1.5B-Instruct,0.000000,reasoning,negation
...,...,...,...,...,...
1543,zorro-quantifiers-superlative,Qwen/Qwen2.5-3B-Instruct,0.967180,semantics,quantifiers
1544,zorro-quantifiers-superlative,Qwen/Qwen2.5-3B-Instruct-AWQ,0.944393,semantics,quantifiers
1545,zorro-quantifiers-superlative,Qwen/Qwen2.5-7B,0.528964,semantics,quantifiers
1546,zorro-quantifiers-superlative,Qwen/Qwen2.5-7B-Instruct,0.722753,semantics,quantifiers


In [436]:
qwen_results_og = qwen_results_og.merge(
    mapping_subfield,
    how='left',
    left_on='probing_dataset',
    right_on='probing dataset'
).merge(
    mapping_phenomena,
    how='left',
    left_on='probing_dataset',
    right_on='probing dataset'
).rename(columns={'probing_dataset': 'probe'}).drop(columns=['Unnamed: 0', 'probing dataset_x', 'probing dataset_y'])
qwen_results_og['model_name'] = qwen_results_og['model_name'].str.replace('__', '/')
qwen_results_og

,probe,model_name,score,linguistic subfield,linguistic phenomena
0,SemAntoNeg,Qwen/Qwen2.5-0.5B,0.002807,reasoning,negation
1,SemAntoNeg,Qwen/Qwen2.5-0.5B-Instruct,0.004167,reasoning,negation
2,SemAntoNeg,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.000000,reasoning,negation
3,SemAntoNeg,Qwen/Qwen2.5-1.5B,0.000000,reasoning,negation
4,SemAntoNeg,Qwen/Qwen2.5-1.5B-Instruct,0.000000,reasoning,negation
...,...,...,...,...,...
1950,zorro-quantifiers-superlative,Qwen/Qwen2.5-3B-Instruct,0.967180,semantics,quantifiers
1951,zorro-quantifiers-superlative,Qwen/Qwen2.5-3B-Instruct-AWQ,0.944393,semantics,quantifiers
1952,zorro-quantifiers-superlative,Qwen/Qwen2.5-7B,0.528964,semantics,quantifiers
1953,zorro-quantifiers-superlative,Qwen/Qwen2.5-7B-Instruct,0.722753,semantics,quantifiers


In [437]:
qwen_results_halfp['model_name'] = qwen_results_halfp['model_name'].str.replace('__', '/')
qwen_results_4bit['model_name'] = qwen_results_4bit['model_name'].str.replace('__', '/')
qwen_results_8bit['model_name'] = qwen_results_8bit['model_name'].str.replace('__', '/')

In [438]:
other_df = other_models.drop(columns=['Unnamed: 0', 'probing dataset', 'linguistic subfield', 'linguistic phenomena', 'probe type', 'train portion']).melt(id_vars=["probe"], var_name="model_name", value_name="score").groupby(["probe", "model_name"], as_index=False).mean()

other_subfield = other_models.drop(columns=['Unnamed: 0', 'probing dataset', 'probe', 'linguistic phenomena', 'probe type', 'train portion']).melt(id_vars=["linguistic subfield"], var_name="model_name", value_name="score").groupby([ 'linguistic subfield', "model_name"], as_index=False).mean()

other_phenomena = other_models.drop(columns=['Unnamed: 0', 'probing dataset', 'probe', 'linguistic subfield', 'probe type', 'train portion']).melt(id_vars=["linguistic phenomena"], var_name="model_name", value_name="score").groupby([ 'linguistic phenomena', "model_name"], as_index=False).mean()


In [439]:
qwen_subfield = qwen_results.drop(columns=['linguistic phenomena']).groupby([ 'linguistic subfield', "model_name"], as_index=False)['score'].mean()
qwen_phenomena = qwen_results.drop(columns=['linguistic subfield']).groupby([ 'linguistic phenomena', "model_name"], as_index=False)['score'].mean()

qwen_subfield_og = qwen_results_og.drop(columns=['linguistic phenomena']).groupby([ 'linguistic subfield', "model_name"], as_index=False)['score'].mean()
qwen_phenomena_og = qwen_results_og.drop(columns=['linguistic subfield']).groupby([ 'linguistic phenomena', "model_name"], as_index=False)['score'].mean()

In [440]:
final_set_sorted = pd.concat([qwen_results, other_df]).sort_values(by=['probe', 'score'], ascending=[True, True])
final_set_subfield_sorted = pd.concat([qwen_subfield, other_subfield]).sort_values(by=['linguistic subfield', 'score'], ascending=[True, True])
final_set_phenomena_sorted = pd.concat([qwen_phenomena, other_phenomena]).sort_values(by=['linguistic phenomena', 'score'], ascending=[True, True])

In [441]:
aa = final_set_phenomena_sorted[final_set_phenomena_sorted['model_name'].str.contains('Qwen')]
aa

,linguistic phenomena,model_name,score
6,age comparison,Qwen/Qwen2.5-3B,0.337576
8,age comparison,Qwen/Qwen2.5-3B-Instruct-AWQ,0.346667
7,age comparison,Qwen/Qwen2.5-3B-Instruct,0.354422
2,age comparison,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.355952
1,age comparison,Qwen/Qwen2.5-0.5B-Instruct,0.377381
...,...,...,...
573,word content,Qwen/Qwen2.5-7B,0.001302
574,word content,Qwen/Qwen2.5-7B-Instruct,0.001696
575,word content,Qwen/Qwen2.5-7B-Instruct-AWQ,0.001893
572,word content,Qwen/Qwen2.5-3B-Instruct-AWQ,0.002456


In [442]:
ab = final_set_subfield_sorted[final_set_subfield_sorted['model_name'].str.contains('Qwen')]
ab

,linguistic subfield,model_name,score
2,discourse,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.284397
0,discourse,Qwen/Qwen2.5-0.5B,0.285105
3,discourse,Qwen/Qwen2.5-1.5B,0.287615
5,discourse,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.289310
1,discourse,Qwen/Qwen2.5-0.5B-Instruct,0.289935
8,discourse,Qwen/Qwen2.5-3B-Instruct-AWQ,0.290815
6,discourse,Qwen/Qwen2.5-3B,0.291462
4,discourse,Qwen/Qwen2.5-1.5B-Instruct,0.293605
7,discourse,Qwen/Qwen2.5-3B-Instruct,0.293917
9,discourse,Qwen/Qwen2.5-7B,0.295196


In [443]:
final_set_sorted_qwen_halfp = pd.concat([qwen_results_halfp, other_df]).sort_values(by=['probe', 'score'], ascending=[True, True])
final_set_sorted_qwen_4bit = pd.concat([qwen_results_4bit, other_df]).sort_values(by=['probe', 'score'], ascending=[True, True])
final_set_sorted_qwen_8bit = pd.concat([qwen_results_8bit, other_df]).sort_values(by=['probe', 'score'], ascending=[True, True])


In [444]:
fig = px.bar(
    final_set_sorted,
    x='score',
    y='model_name',
    color='score',
    animation_frame='probe',
    title='Rankings of Models for Each Probe',
    labels={'model_name': 'Model Name', 'score': 'Score'},
    height=1000,
    width=1300
)

steps = [
    dict(
        label=probe,
        method="animate",
        args=[
            [probe],
            dict(
                mode="immediate",
                frame={"duration": 500, "redraw": True},
                transition={"duration": 300},
            ),
        ],
    )
    for probe in final_set_sorted['probe'].unique()
]

fig.update_layout(
    xaxis_title="Score",
    yaxis_title="Model Name",
    title_x=0.5,
    margin=dict(l=150, r=40, t=80, b=80), 
    sliders=[
        dict(
            active=0,
            pad={"t": 50},  
            len=0.9, 
            currentvalue={"font": {"size": 20}},  
            steps=steps,
        )
    ]
)

fig["layout"].pop("updatemenus")
fig.show()

In [445]:
# Create the bar chart
fig = px.bar(
    final_set_subfield_sorted,
    x='score',
    y='model_name',
    color='score',
    animation_frame='linguistic subfield',
    title='Rankings of Models for each Linguistic Subfield',
    labels={'model_name': 'Model Name', 'score': 'Score'},
    height=1000,
    width=1300
)

steps = [
    dict(
        label=subfield,
        method="animate",
        args=[
            [subfield],
            dict(
                mode="immediate",
                frame={"duration": 500, "redraw": True},
                transition={"duration": 300},
            ),
        ],
    )
    for subfield in final_set_sorted['linguistic subfield'].unique()
]

fig.update_layout(
    xaxis_title="Score",
    yaxis_title="Model Name",
    title_x=0.5,
    margin=dict(l=150, r=40, t=80, b=80),  # Add space for slider
    sliders=[
        dict(
            active=0,
            pad={"t": 50},  # Padding to move slider down
            len=0.9,  # Length of the slider
            currentvalue={"font": {"size": 20}},  # Larger font for slider value
            steps=steps,
        )
    ],
)

# Show the updated plot
fig["layout"].pop("updatemenus")
fig.show()

In [446]:
# Create the bar chart
fig = px.bar(
    final_set_phenomena_sorted,
    x='score',
    y='model_name',
    color='score',
    animation_frame='linguistic phenomena',
    title='Rankings of Models for each Linguistic Phenomena',
    labels={'model_name': 'Model Name', 'score': 'Score'},
    height=1000,
    width=1300
)

steps = [
    dict(
        label=phenomena,
        method="animate",
        args=[
            [phenomena],
            dict(
                mode="immediate",
                frame={"duration": 500, "redraw": True},
                transition={"duration": 300},
            ),
        ],
    )
    for phenomena in final_set_sorted['linguistic phenomena'].unique()
]

fig.update_layout(
    xaxis_title="Score",
    yaxis_title="Model Name",
    title_x=0.5,
    margin=dict(l=150, r=40, t=80, b=80),  # Add space for slider
    sliders=[
        dict(
            active=0,
            pad={"t": 50},  # Padding to move slider down
            len=0.9,  # Length of the slider
            currentvalue={"font": {"size": 20}},  # Larger font for slider value
            steps=steps,
        )
    ],
)

# Show the updated plot
fig["layout"].pop("updatemenus")
fig.show()

In [447]:
import pandas as pd
import plotly.express as px

def generate_custom_plots(dataframe, column_name='linguistic subfield', n=15):
    """
    Generates custom plots for top and bottom n models grouped by a specified column.
    
    Parameters:
    - dataframe: pd.DataFrame - The input dataframe containing model data.
    - column_name: str - The column to group by (replaces 'linguistic subfield').
    - n: int - The number of top and bottom entries to filter for each group.
    
    Returns:
    - None: Displays the generated plots.
    """
    def filter_top_bottom(df, top=True, n=n):
        filtered_dfs = []
        for group in df[column_name].unique():
            group_df = df[df[column_name] == group]
            if top:
                filtered_dfs.append(group_df.nlargest(n, 'score'))
            else:
                filtered_dfs.append(group_df.nsmallest(n, 'score'))
        return pd.concat(filtered_dfs)

    # Filter top and bottom models
    top_models = filter_top_bottom(dataframe, top=True, n=n)
    bottom_models = filter_top_bottom(dataframe, top=False, n=n)

    # Create steps for arrow key control
    def create_steps(df):
        return [
            dict(
                label=group,
                method="animate",
                args=[
                    [group],
                    dict(
                        mode="immediate",
                        frame={"duration": 500, "redraw": True},
                        transition={"duration": 300},
                    ),
                ],
            )
            for group in df[column_name].unique()
        ]

    # Plot for Top Models
    fig_top = px.bar(
        top_models,
        x='score',
        y='model_name',
        color='score',
        animation_frame=column_name,
        title=f'Top {n} Models for Each {column_name.capitalize()}',
        labels={'model_name': 'Model Name', 'score': 'Score'},
        height=1000,
        width=1300
    )

    fig_top.update_layout(
        xaxis_title="Score",
        yaxis_title="Model Name",
        title_x=0.5,
        margin=dict(l=150, r=40, t=80, b=80),
        yaxis=dict(
            categoryorder='total ascending'  # Sort highest score at the top
        ),
        sliders=[
            dict(
                active=0,
                pad={"t": 50},
                len=0.9,
                currentvalue={"font": {"size": 20}},
                steps=create_steps(top_models)  # Add steps for arrow key navigation
            )
        ]
    )

    # Plot for Bottom Models
    fig_bottom = px.bar(
        bottom_models,
        x='score',
        y='model_name',
        color='score',
        animation_frame=column_name,
        title=f'Bottom {n} Models for Each {column_name.capitalize()}',
        labels={'model_name': 'Model Name', 'score': 'Score'},
        height=1000,
        width=1300
    )

    fig_bottom.update_layout(
        xaxis_title="Score",
        yaxis_title="Model Name",
        title_x=0.5,
        margin=dict(l=150, r=40, t=80, b=80),
        yaxis=dict(
            categoryorder='total descending'  # Sort lowest score at the top
        ),
        sliders=[
            dict(
                active=0,
                pad={"t": 50},
                len=0.9,
                currentvalue={"font": {"size": 20}},
                steps=create_steps(bottom_models)  # Add steps for arrow key navigation
            )
        ]
    )

    # Ensure unnecessary buttons are removed
    fig_top["layout"].pop("updatemenus", None)
    fig_bottom["layout"].pop("updatemenus", None)

    # Show both plots
    fig_top.show()
    fig_bottom.show()

In [448]:
generate_custom_plots(final_set_sorted, 'probe', 15)

In [449]:
generate_custom_plots(final_set_subfield_sorted, 'linguistic subfield', 15)

In [450]:
generate_custom_plots(final_set_phenomena_sorted, 'linguistic phenomena', 15)

# Statistical Calculation
Null Hypothesis = There is 0 difference between quantized and non-quantized model 

Alternative Hypothesis = There is some difference between quantized and non-quantized model, to be exact the quantized model has got a worse performance than the unquantized one

p-value < 0.05 => Rejection of our Null Hypothesis = Statistical Significance

In [451]:
final_set_sorted['model_name'].unique().tolist()

['Qwen/Qwen2.5-0.5B-Instruct-AWQ',
 'Qwen/Qwen2.5-1.5B',
 'Qwen/Qwen2.5-1.5B-Instruct',
 'Qwen/Qwen2.5-1.5B-Instruct-AWQ',
 'Qwen/Qwen2.5-3B',
 'Qwen/Qwen2.5-3B-Instruct',
 'Qwen/Qwen2.5-3B-Instruct-AWQ',
 'Qwen/Qwen2.5-7B',
 'Qwen/Qwen2.5-7B-Instruct',
 'Qwen/Qwen2.5-7B-Instruct-AWQ',
 'Qwen/Qwen2.5-0.5B',
 'Qwen/Qwen2.5-0.5B-Instruct',
 'EleutherAI/pythia-70m',
 'EleutherAI/pythia-160m',
 'EleutherAI/pythia-160m-deduped',
 'EleutherAI/pythia-70m-deduped',
 'mistralai/Mixtral-8x7B-v0.1',
 'google/t5-small-lm-adapt',
 'EleutherAI/pythia-410m-deduped',
 'ibm/merlinite-7b',
 'google/t5-large-lm-adapt',
 'mistralai/Mixtral-8x7B-Instruct-v0.1',
 'gpt2',
 'mistralai/Mistral-7B-v0.1',
 'google/flan-t5-small',
 'mistralai/Mistral-7B-Instruct-v0.1',
 'google/flan-t5-large',
 'google/t5-xl-lm-adapt',
 'roberta-base',
 'EleutherAI/pythia-410m',
 'google/t5-base-lm-adapt',
 'google/electra-base-discriminator',
 'sentence-transformers/average_word_embeddings_glove.840B.300d',
 'meta-llama/Llama-2-

## Only Sentence and Sentence Pair Classification (Probe Type)

In [452]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-0.5B-Instruct-AWQ"
model_2 = "Qwen/Qwen2.5-0.5B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted[final_set_sorted['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significan t difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -0.00750532163818586
P-value: 0.994017521284021
There is no statistically significant difference between the two models.


In [453]:
from scipy.stats import levene

# Perform Levene's test
stat, p_value = levene(model_1_scores, model_2_scores)

# Display the results
print(f"Levene's Test Statistic: {stat}")
print(f"P-value: {p_value}")

if p_value < 0.05:
    print("Variances are significantly different (heteroscedasticity).")
else:
    print("No significant difference in variances (homoscedasticity).")

Levene's Test Statistic: 0.03080965572430462
P-value: 0.86080447502539
No significant difference in variances (homoscedasticity).


In [454]:
from scipy.stats import bartlett

# Perform Bartlett's test
stat, p_value = bartlett(model_1_scores, model_2_scores)

# Display the results
print(f"Bartlett's Test Statistic: {stat}")
print(f"P-value: {p_value}")

if p_value < 0.05:
    print("Variances are significantly different (heteroscedasticity).")
else:
    print("No significant difference in variances (homoscedasticity).")

Bartlett's Test Statistic: 0.006679420620008238
P-value: 0.9348632327548473
No significant difference in variances (homoscedasticity).


In [455]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-1.5B-Instruct-AWQ"
model_2 = "Qwen/Qwen2.5-1.5B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted[final_set_sorted['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: 0.184484922075762
P-value: 0.8537791132226715
There is no statistically significant difference between the two models.


In [456]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-3B-Instruct-AWQ"
model_2 = "Qwen/Qwen2.5-3B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted[final_set_sorted['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -0.019446969779903903
P-value: 0.984499689212486
There is no statistically significant difference between the two models.


In [457]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-7B-Instruct-AWQ"
model_2 = "Qwen/Qwen2.5-7B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted[final_set_sorted['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: 1.0699756532255544
P-value: 0.28563805302398654
There is no statistically significant difference between the two models.


In [458]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-3B-Instruct-AWQ"
model_2 = "Qwen/Qwen2.5-3B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_subfield_sorted[final_set_subfield_sorted['model_name'] == model_1]['score']
model_2_scores = final_set_subfield_sorted[final_set_subfield_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -0.02062848992542131
P-value: 0.9840472391083873
There is no statistically significant difference between the two models.


In [459]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-3B-Instruct-AWQ"
model_2 = "Qwen/Qwen2.5-3B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_phenomena_sorted[final_set_phenomena_sorted['model_name'] == model_1]['score']
model_2_scores = final_set_phenomena_sorted[final_set_phenomena_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: 0.0003511323964695852
P-value: 0.999720580998556
There is no statistically significant difference between the two models.


# Entire Data of Qwen Models
This calculation uses the entire results.csv, with all probe types instead of just sentence and sentence pair classification

## AWQ models (4-bit quantization)

In [460]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-3B-Instruct-AWQ"
model_2 = "Qwen/Qwen2.5-3B-Instruct"

# Filter the data for the two models
model_1_scores = qwen_results_og[qwen_results_og['model_name'] == model_1]['score']
model_2_scores = qwen_results_og[qwen_results_og['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores, equal_var=False)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -3.2876971893181857
P-value: 0.0011178935690396821
There is a statistically significant difference between the two models.


In [461]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-3B-Instruct-AWQ"
model_2 = "Qwen/Qwen2.5-3B-Instruct"

# Filter the data for the two models
model_1_scores = qwen_subfield_og[qwen_subfield_og['model_name'] == model_1]['score']
model_2_scores = qwen_subfield_og[qwen_subfield_og['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores, equal_var=False)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -0.5733685524165013
P-value: 0.5822087123033508
There is no statistically significant difference between the two models.


In [462]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-3B-Instruct-AWQ"
model_2 = "Qwen/Qwen2.5-3B-Instruct"

# Filter the data for the two models
model_1_scores = qwen_phenomena_og[qwen_phenomena_og['model_name'] == model_1]['score']
model_2_scores = qwen_phenomena_og[qwen_phenomena_og['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores, equal_var=False)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -1.058339998958577
P-value: 0.2924075305056538
There is no statistically significant difference between the two models.


## Half Precision Models (Bitsandbytes)

In [463]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-0.5B-Instruct"
model_2 = "Qwen/Qwen2.5-0.5B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted_qwen_halfp[final_set_sorted_qwen_halfp['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -0.9028131203679151
P-value: 0.3674737728861479
There is no statistically significant difference between the two models.


In [464]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-1.5B-Instruct"
model_2 = "Qwen/Qwen2.5-1.5B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted_qwen_halfp[final_set_sorted_qwen_halfp['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -0.19535401003939973
P-value: 0.8452707784389817
There is no statistically significant difference between the two models.


In [465]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-3B-Instruct"
model_2 = "Qwen/Qwen2.5-3B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted_qwen_halfp[final_set_sorted_qwen_halfp['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -0.03248014476037531
P-value: 0.9741144587803985
There is no statistically significant difference between the two models.


In [466]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-7B-Instruct"
model_2 = "Qwen/Qwen2.5-7B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted_qwen_halfp[final_set_sorted_qwen_halfp['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -0.4743308814301463
P-value: 0.6356683075194303
There is no statistically significant difference between the two models.


## 8Bit Precision Models (Bitsandbytes)

In [467]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-0.5B-Instruct"
model_2 = "Qwen/Qwen2.5-0.5B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted_qwen_8bit[final_set_sorted_qwen_8bit['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -0.8899186641499189
P-value: 0.3743450819198023
There is no statistically significant difference between the two models.


In [468]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-1.5B-Instruct"
model_2 = "Qwen/Qwen2.5-1.5B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted_qwen_8bit[final_set_sorted_qwen_8bit['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -0.43289110426016036
P-value: 0.6654583622190078
There is no statistically significant difference between the two models.


In [469]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-3B-Instruct"
model_2 = "Qwen/Qwen2.5-3B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted_qwen_8bit[final_set_sorted_qwen_8bit['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -0.15943649825544873
P-value: 0.8734507018767175
There is no statistically significant difference between the two models.


In [470]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-7B-Instruct"
model_2 = "Qwen/Qwen2.5-7B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted_qwen_8bit[final_set_sorted_qwen_8bit['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -0.3541489906902324
P-value: 0.723518750666287
There is no statistically significant difference between the two models.


## 4Bit Precision Models (Bitsandbytes)

In [471]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-0.5B-Instruct"
model_2 = "Qwen/Qwen2.5-0.5B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted_qwen_4bit[final_set_sorted_qwen_4bit['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -0.6812515611851304
P-value: 0.49632810612662304
There is no statistically significant difference between the two models.


In [472]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-1.5B-Instruct"
model_2 = "Qwen/Qwen2.5-1.5B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted_qwen_4bit[final_set_sorted_qwen_4bit['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: 0.044059498333069005
P-value: 0.9648913239141064
There is no statistically significant difference between the two models.


In [473]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-3B-Instruct"
model_2 = "Qwen/Qwen2.5-3B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted_qwen_4bit[final_set_sorted_qwen_4bit['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: -0.8340891306813425
P-value: 0.4050083710289145
There is no statistically significant difference between the two models.


In [474]:
# Specify the two models you want to compare
model_1 = "Qwen/Qwen2.5-7B-Instruct"
model_2 = "Qwen/Qwen2.5-7B-Instruct"

# Filter the data for the two models
model_1_scores = final_set_sorted_qwen_4bit[final_set_sorted_qwen_4bit['model_name'] == model_1]['score']
model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']

# Perform the t-test
t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpret the result
if p_value < 0.05:
    print("There is a statistically significant difference between the two models.")
else:
    print("There is no statistically significant difference between the two models.")

T-statistic: 0.04608442370639275
P-value: 0.9632788895589937
There is no statistically significant difference between the two models.


----------

# Statiscial T-Tests between all model combinations

In [475]:
from itertools import combinations
from scipy.stats import ttest_ind
import pandas as pd


models = final_set_sorted['model_name'].unique()

results = []

for model_1, model_2 in combinations(models, 2):
    model_1_scores = final_set_sorted[final_set_sorted['model_name'] == model_1]['score']
    model_2_scores = final_set_sorted[final_set_sorted['model_name'] == model_2]['score']
    
    t_stat, p_value = ttest_ind(model_1_scores, model_2_scores)
    
    results.append({
        'Model 1': model_1,
        'Model 2': model_2,
        'T-Statistic': t_stat,
        'P-Value': p_value,
        'Significant': p_value < 0.05
    })

results_df = pd.DataFrame(results)

results_df = results_df.sort_values(by='P-Value')

In [476]:
results_df

,Model 1,Model 2,T-Statistic,P-Value,Significant
999,EleutherAI/pythia-70m-deduped,meta-llama/Llama-2-70b-chat-hf,-23.010543,2.879302e-64,True
982,EleutherAI/pythia-70m-deduped,google/flan-t5-xxl,-23.010191,2.886920e-64,True
831,EleutherAI/pythia-70m,meta-llama/Llama-2-70b-chat-hf,-22.835248,1.076581e-63,True
814,EleutherAI/pythia-70m,google/flan-t5-xxl,-22.834355,1.083848e-63,True
888,EleutherAI/pythia-160m,meta-llama/Llama-2-70b-chat-hf,-22.533412,1.051533e-62,True
...,...,...,...,...,...
1737,google/electra-base-discriminator,microsoft/Orca-2-13b,0.006512,9.948097e-01,False
2147,EleutherAI/pythia-1.4b-deduped,bert-base-uncased,-0.005252,9.958135e-01,False
2306,databricks/dolly-v2-12b,meta-llama/Llama-2-13b-chat-hf,-0.003912,9.968818e-01,False
1800,meta-llama/Llama-2-7b-hf,EleutherAI/pythia-1b,0.003541,9.971775e-01,False


In [477]:
results_df[(results_df['Model 1'].str.contains("Qwen")) &  (results_df['Model 2'].str.contains("Qwen"))]

,Model 1,Model 2,T-Statistic,P-Value,Significant
408,Qwen/Qwen2.5-3B-Instruct-AWQ,Qwen/Qwen2.5-0.5B,5.608421,5.284029e-08,True
344,Qwen/Qwen2.5-3B-Instruct,Qwen/Qwen2.5-0.5B,5.599381,5.535401e-08,True
279,Qwen/Qwen2.5-3B,Qwen/Qwen2.5-0.5B,5.589389,5.826777e-08,True
213,Qwen/Qwen2.5-1.5B-Instruct-AWQ,Qwen/Qwen2.5-0.5B,5.261084,3.028221e-07,True
5,Qwen/Qwen2.5-0.5B-Instruct-AWQ,Qwen/Qwen2.5-3B-Instruct-AWQ,-5.190148,4.281711e-07,True
...,...,...,...,...,...
145,Qwen/Qwen2.5-1.5B-Instruct,Qwen/Qwen2.5-7B-Instruct-AWQ,-0.034929,9.721633e-01,False
275,Qwen/Qwen2.5-3B,Qwen/Qwen2.5-3B-Instruct-AWQ,0.022415,9.821343e-01,False
340,Qwen/Qwen2.5-3B-Instruct,Qwen/Qwen2.5-3B-Instruct-AWQ,0.019447,9.844997e-01,False
10,Qwen/Qwen2.5-0.5B-Instruct-AWQ,Qwen/Qwen2.5-0.5B-Instruct,-0.007505,9.940175e-01,False


-------

# Rankings

In [478]:
exclude_models_instruct_and_base = [
    f"Qwen/Qwen2.5-{param}B-Instruct"
    for param in [3, 1.5, 7, 0.5]
] + [
    f"Qwen/Qwen2.5-{param}B"
    for param in [3, 1.5, 7, 0.5]
]

exclude_models_awq = [
    f"Qwen/Qwen2.5-{param}B-Instruct-AWQ"
    for param in [3, 1.5, 7, 0.5]
]

rankings_with_awq = (
    final_set_sorted[
        ~final_set_sorted['model_name'].isin(exclude_models_instruct_and_base)
    ]
    .groupby(['model_name'])['score']
    .mean()
    .sort_values(ascending=False)
    .to_frame().reset_index()
)

rankings_with_awq['rank'] = rankings_with_awq['score'].rank(ascending=False, method='min').astype(int)

rankings_without_awq = (
    final_set_sorted[
        ~final_set_sorted['model_name'].isin(exclude_models_awq)
    ]
    .groupby(['model_name'])['score']
    .mean()
    .sort_values(ascending=False)
    .to_frame().reset_index()
)

rankings_without_awq['rank'] = rankings_without_awq['score'].rank(ascending=False, method='min').astype(int)

In [479]:
rankings_with_awq_halfp = (
    final_set_sorted_qwen_halfp[
        ~final_set_sorted_qwen_halfp['model_name'].isin(exclude_models_instruct_and_base)
    ]
    .groupby(['model_name'])['score']
    .mean()
    .sort_values(ascending=False)
    .to_frame().reset_index()
)

rankings_with_awq_halfp['rank'] = rankings_with_awq_halfp['score'].rank(ascending=False, method='min').astype(int)

rankings_without_awq_halfp = (
    final_set_sorted_qwen_halfp[
        ~final_set_sorted_qwen_halfp['model_name'].isin(exclude_models_awq)
    ]
    .groupby(['model_name'])['score']
    .mean()
    .sort_values(ascending=False)
    .to_frame().reset_index()
)

rankings_without_awq_halfp['rank'] = rankings_without_awq_halfp['score'].rank(ascending=False, method='min').astype(int)

In [480]:
rankings_with_awq_4bit = (
    final_set_sorted_qwen_4bit[
        ~final_set_sorted_qwen_4bit['model_name'].isin(exclude_models_instruct_and_base)
    ]
    .groupby(['model_name'])['score']
    .mean()
    .sort_values(ascending=False)
    .to_frame().reset_index()
)

rankings_with_awq_4bit['rank'] = rankings_with_awq_4bit['score'].rank(ascending=False, method='min').astype(int)

rankings_without_awq_4bit = (
    final_set_sorted_qwen_4bit[
        ~final_set_sorted_qwen_4bit['model_name'].isin(exclude_models_awq)
    ]
    .groupby(['model_name'])['score']
    .mean()
    .sort_values(ascending=False)
    .to_frame().reset_index()
)

rankings_without_awq_4bit['rank'] = rankings_without_awq_4bit['score'].rank(ascending=False, method='min').astype(int)

In [481]:
rankings_with_awq_8bit = (
    final_set_sorted_qwen_8bit[
        ~final_set_sorted_qwen_8bit['model_name'].isin(exclude_models_instruct_and_base)
    ]
    .groupby(['model_name'])['score']
    .mean()
    .sort_values(ascending=False)
    .to_frame().reset_index()
)

rankings_with_awq_8bit['rank'] = rankings_with_awq_8bit['score'].rank(ascending=False, method='min').astype(int)

rankings_without_awq_8bit = (
    final_set_sorted_qwen_8bit[
        ~final_set_sorted_qwen_8bit['model_name'].isin(exclude_models_awq)
    ]
    .groupby(['model_name'])['score']
    .mean()
    .sort_values(ascending=False)
    .to_frame().reset_index()
)

rankings_without_awq_8bit['rank'] = rankings_without_awq_8bit['score'].rank(ascending=False, method='min').astype(int)

In [482]:
print("Rankings including AWQ models but excluding other Qwen models:")
print(rankings_with_awq)

Rankings including AWQ models but excluding other Qwen models:
                                           model_name     score  rank
0                               lmsys/vicuna-13b-v1.5  0.741246     1
1                      meta-llama/Llama-2-70b-chat-hf  0.739794     2
2                                  google/flan-t5-xxl  0.738153     3
3                              allenai/tulu-2-dpo-13b  0.736772     4
4                                  allenai/tulu-2-13b  0.736452     5
..                                                ...       ...   ...
58  sentence-transformers/average_word_embeddings_...  0.510593    59
59                     EleutherAI/pythia-160m-deduped  0.352659    60
60                             EleutherAI/pythia-160m  0.341424    61
61                              EleutherAI/pythia-70m  0.336837    62
62                      EleutherAI/pythia-70m-deduped  0.333009    63

[63 rows x 3 columns]


In [483]:
print("\nRankings excluding AWQ models:")
print(rankings_without_awq.to_latex())


Rankings excluding AWQ models:
                                           model_name     score  rank
0                               lmsys/vicuna-13b-v1.5  0.741246     1
1                      meta-llama/Llama-2-70b-chat-hf  0.739794     2
2                                  google/flan-t5-xxl  0.738153     3
3                              allenai/tulu-2-dpo-13b  0.736772     4
4                                  allenai/tulu-2-13b  0.736452     5
..                                                ...       ...   ...
62  sentence-transformers/average_word_embeddings_...  0.510593    63
63                     EleutherAI/pythia-160m-deduped  0.352659    64
64                             EleutherAI/pythia-160m  0.341424    65
65                              EleutherAI/pythia-70m  0.336837    66
66                      EleutherAI/pythia-70m-deduped  0.333009    67

[67 rows x 3 columns]


In [537]:
latex_context = rankings_without_awq.to_latex(escape=True, position='H', longtable=True, index=False)
with open('Table2.tex', "w") as file:
    file.write(latex_context)

In [484]:
rankings_with_awq_halfp

,model_name,score,rank
0,lmsys/vicuna-13b-v1.5,0.741246,1
1,meta-llama/Llama-2-70b-chat-hf,0.739794,2
2,google/flan-t5-xxl,0.738153,3
3,allenai/tulu-2-dpo-13b,0.736772,4
4,allenai/tulu-2-13b,0.736452,5
...,...,...,...
57,sentence-transformers/average_word_embeddings_...,0.510593,58
58,EleutherAI/pythia-160m-deduped,0.352659,59
59,EleutherAI/pythia-160m,0.341424,60
60,EleutherAI/pythia-70m,0.336837,61


In [485]:
rankings_without_awq_halfp

,model_name,score,rank
0,lmsys/vicuna-13b-v1.5,0.741246,1
1,meta-llama/Llama-2-70b-chat-hf,0.739794,2
2,google/flan-t5-xxl,0.738153,3
3,allenai/tulu-2-dpo-13b,0.736772,4
4,allenai/tulu-2-13b,0.736452,5
...,...,...,...
58,sentence-transformers/average_word_embeddings_...,0.510593,59
59,EleutherAI/pythia-160m-deduped,0.352659,60
60,EleutherAI/pythia-160m,0.341424,61
61,EleutherAI/pythia-70m,0.336837,62


In [486]:
qwen_no_awq = rankings_without_awq[rankings_without_awq['model_name'].isin(exclude_models_instruct_and_base)]
qwen_no_awq

,model_name,score,rank
22,Qwen/Qwen2.5-3B,0.707795,23
23,Qwen/Qwen2.5-3B-Instruct,0.707698,24
32,Qwen/Qwen2.5-1.5B-Instruct,0.691648,33
34,Qwen/Qwen2.5-1.5B,0.689122,35
45,Qwen/Qwen2.5-7B-Instruct,0.658552,46
49,Qwen/Qwen2.5-7B,0.640303,50
58,Qwen/Qwen2.5-0.5B-Instruct,0.548603,59
61,Qwen/Qwen2.5-0.5B,0.534483,62


In [487]:
qwen_normal_rankings = qwen_no_awq[~qwen_no_awq['model_name'].str.contains('Instruct')]

In [488]:
qwen_instruct_rankings = qwen_no_awq[qwen_no_awq['model_name'].str.contains('Instruct')]

In [489]:
qwen_awq_rankings = rankings_with_awq[rankings_with_awq['model_name'].isin(exclude_models_awq)]

In [490]:
def calculate_average_rank_difference(df1, df2, id_column, rank_column):
    merged = pd.merge(df1, df2, on=id_column)
    merged['Rank_Difference'] = abs(merged[f'{rank_column}_x'] - merged[f'{rank_column}_y'])
    return merged['Rank_Difference'].mean()

In [535]:
qwen_instruct_rankings

,model_name,score,rank
23,Qwen/Qwen2.5-3B-Instruct,0.707698,24
32,Qwen/Qwen2.5-1.5B-Instruct,0.691648,33
45,Qwen/Qwen2.5-7B-Instruct,0.658552,46
58,Qwen/Qwen2.5-0.5B-Instruct,0.548603,59


In [492]:
qwen_awq_rankings

,model_name,score,rank
23,Qwen/Qwen2.5-3B-Instruct-AWQ,0.707068,24
30,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.697668,31
32,Qwen/Qwen2.5-7B-Instruct-AWQ,0.692771,33
55,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.548386,56


Average Ranking difference between Instruct and AWQ Instruct model is around 6.25, with AWQ models actually being better overall, which is surprising...

In [493]:
(((24-24)+(33-31)+(46-33)+(59-56)))/4

4.5

In [494]:
qwen_normal_rankings

,model_name,score,rank
22,Qwen/Qwen2.5-3B,0.707795,23
34,Qwen/Qwen2.5-1.5B,0.689122,35
49,Qwen/Qwen2.5-7B,0.640303,50
61,Qwen/Qwen2.5-0.5B,0.534483,62


Ranking difference of 3.3333 ranks with normal and quantized models

In [495]:
((24-23)+(35-31)+(50-33)+(62-56))/4

7.0

# Half Precision Ranking

In [496]:
qwen_no_awq_halfp = rankings_without_awq_halfp[rankings_without_awq_halfp['model_name'].isin(exclude_models_instruct_and_base)]

In [497]:
qwen_no_awq_halfp

,model_name,score,rank
23,Qwen/Qwen2.5-3B-Instruct,0.706645,24
35,Qwen/Qwen2.5-1.5B-Instruct,0.685317,36
46,Qwen/Qwen2.5-7B-Instruct,0.643142,47
57,Qwen/Qwen2.5-0.5B-Instruct,0.522341,58


In [498]:
qwen_instruct_rankings

,model_name,score,rank
23,Qwen/Qwen2.5-3B-Instruct,0.707698,24
32,Qwen/Qwen2.5-1.5B-Instruct,0.691648,33
45,Qwen/Qwen2.5-7B-Instruct,0.658552,46
58,Qwen/Qwen2.5-0.5B-Instruct,0.548603,59


In [499]:
calculate_average_rank_difference(qwen_instruct_rankings, qwen_no_awq_halfp , 'model_name', 'rank')

np.float64(1.25)

# 8Bit Precision Ranking

In [500]:
qwen_no_awq_8bit = rankings_without_awq_8bit[rankings_without_awq_8bit['model_name'].isin(exclude_models_instruct_and_base)]

In [501]:
qwen_no_awq_8bit

,model_name,score,rank
25,Qwen/Qwen2.5-3B-Instruct,0.702523,26
37,Qwen/Qwen2.5-1.5B-Instruct,0.677546,38
46,Qwen/Qwen2.5-7B-Instruct,0.647119,47
57,Qwen/Qwen2.5-0.5B-Instruct,0.522933,58


In [502]:
qwen_instruct_rankings

,model_name,score,rank
23,Qwen/Qwen2.5-3B-Instruct,0.707698,24
32,Qwen/Qwen2.5-1.5B-Instruct,0.691648,33
45,Qwen/Qwen2.5-7B-Instruct,0.658552,46
58,Qwen/Qwen2.5-0.5B-Instruct,0.548603,59


In [503]:
calculate_average_rank_difference(qwen_instruct_rankings, qwen_no_awq_8bit , 'model_name', 'rank')

np.float64(2.25)

# 4Bit Precision Ranking
Bitsandbytes Quantization, not AWQ

In [504]:
qwen_no_awq_4bit = rankings_without_awq_4bit[rankings_without_awq_4bit['model_name'].isin(exclude_models_instruct_and_base)]

In [505]:
qwen_no_awq_4bit

,model_name,score,rank
30,Qwen/Qwen2.5-1.5B-Instruct,0.693088,31
37,Qwen/Qwen2.5-3B-Instruct,0.681069,38
42,Qwen/Qwen2.5-7B-Instruct,0.660045,43
57,Qwen/Qwen2.5-0.5B-Instruct,0.528471,58


In [506]:
qwen_awq_rankings

,model_name,score,rank
23,Qwen/Qwen2.5-3B-Instruct-AWQ,0.707068,24
30,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.697668,31
32,Qwen/Qwen2.5-7B-Instruct-AWQ,0.692771,33
55,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.548386,56


In [507]:
qwen_instruct_rankings

,model_name,score,rank
23,Qwen/Qwen2.5-3B-Instruct,0.707698,24
32,Qwen/Qwen2.5-1.5B-Instruct,0.691648,33
45,Qwen/Qwen2.5-7B-Instruct,0.658552,46
58,Qwen/Qwen2.5-0.5B-Instruct,0.548603,59


In [508]:
calculate_average_rank_difference(qwen_instruct_rankings, qwen_no_awq_4bit , 'model_name', 'rank')

np.float64(5.0)

The higher the quantization, the stronger the rank differences between quantized and unquantized models.
Models with 4-bit quantization have got an average ranking difference between 4.0 and 7.0, the Instruct-AWQ models seem to perform better than the Instruct models having lower rankings compared to the Bitsandbytes 4-bit quantization model, which as expected perform worse (but not too much) than the normal Instruct models.
Models with 8-bit and half-precision (16-bit) perform only slightly worse than their non-quantized Instruct counterparts with a ranking difference of 2.25 and 1.25 respectively.

----

## Test

In [509]:
def calculate_score_change(final_set, model_name_x, model_name_y, column_group='linguistic phenomena'):
    one = final_set[final_set['model_name'] == model_name_x]
    two = final_set[final_set['model_name'] == model_name_y]

    three = pd.merge(one, two, on=column_group)

    three['score_change_pct'] = (three['score_y'] - three['score_x']) / three['score_x'] * 100

    three = three.fillna(0).sort_values(by='score_change_pct')

    return three

# Over all Probes

In [533]:
all_probes_05b = calculate_score_change(
    final_set=final_set_sorted,
    model_name_x='Qwen/Qwen2.5-0.5B-Instruct', 
    model_name_y='Qwen/Qwen2.5-0.5B-Instruct-AWQ',
    column_group='probe'
)
# all_probes_05b = all_probes_05b.drop(['linguistic subfield_x', 'linguistic phenomena_x', 'linguistic subfield_y', 'linguistic phenomena_y'], axis=1)
# all_probes_05b['score_change_pct'] = all_probes_05b['score_change_pct'].apply(lambda s: f"{s:.2f}\\%")
# latex_content = all_probes_05b.to_latex(escape=True, position='H', longtable=True, index=False)
# with open('../../Bericht/assets/tables/AllProbes05b.tex', "w") as file:
#     file.write(latex_content)

In [511]:
all_probes_05b

,probe,model_name_x,score_x,linguistic subfield_x,linguistic phenomena_x,model_name_y,score_y,linguistic subfield_y,linguistic phenomena_y,score_change_pct
0,SemAntoNeg,Qwen/Qwen2.5-0.5B-Instruct,0.004167,reasoning,negation,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.000000,reasoning,negation,-100.000000
93,passive,Qwen/Qwen2.5-0.5B-Instruct,0.172720,semantics,passive,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.097753,semantics,passive,-43.403875
103,senteval-word_content,Qwen/Qwen2.5-0.5B-Instruct,0.000374,semantics,word content,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.000213,semantics,word content,-42.878630
109,zorro-agreement_subject_verb-across_prepositio...,Qwen/Qwen2.5-0.5B-Instruct,0.576943,syntax,subject-verb agreement,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.344395,syntax,subject-verb agreement,-40.307018
120,zorro-filler-gap-wh_question_object,Qwen/Qwen2.5-0.5B-Instruct,0.956328,syntax,filler gap,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.582505,syntax,filler gap,-39.089430
...,...,...,...,...,...,...,...,...,...,...
31,blimp-irregular_past_participle_adjectives,Qwen/Qwen2.5-0.5B-Instruct,0.580661,morphology,irregular forms,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.780969,morphology,irregular forms,34.496378
85,olmpics-antonym_synonym_negation,Qwen/Qwen2.5-0.5B-Instruct,0.400252,reasoning,antonym negation,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.541671,reasoning,antonym negation,35.332521
108,zorro-agreement_determiner_noun-between_neighbors,Qwen/Qwen2.5-0.5B-Instruct,0.404919,morphology,determiner noun agreement,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.549595,morphology,determiner noun agreement,35.729810
92,olmpics-size_comparison,Qwen/Qwen2.5-0.5B-Instruct,0.204155,reasoning,object comparision,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.460664,reasoning,object comparision,125.644215


In [512]:
all_probes_15b = calculate_score_change(
    final_set=final_set_sorted,
    model_name_x='Qwen/Qwen2.5-1.5B-Instruct', 
    model_name_y='Qwen/Qwen2.5-1.5B-Instruct-AWQ',
    column_group='probe'
)
all_probes_15b

,probe,model_name_x,score_x,linguistic subfield_x,linguistic phenomena_x,model_name_y,score_y,linguistic subfield_y,linguistic phenomena_y,score_change_pct
103,senteval-word_content,Qwen/Qwen2.5-1.5B-Instruct,0.000846,semantics,word content,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.000366,semantics,word content,-56.670974
107,synonym-antonym-hard,Qwen/Qwen2.5-1.5B-Instruct,0.314290,semantics,synonym-/antonym-detection,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.233311,semantics,synonym-/antonym-detection,-25.765701
81,discourse-connective,Qwen/Qwen2.5-1.5B-Instruct,0.073738,discourse,discourse connective,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.060835,discourse,discourse connective,-17.498662
92,olmpics-size_comparison,Qwen/Qwen2.5-1.5B-Instruct,0.487882,reasoning,object comparision,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.407964,reasoning,object comparision,-16.380633
101,senteval-top_constituents,Qwen/Qwen2.5-1.5B-Instruct,0.445716,syntax,top-constituent-task,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.373661,syntax,top-constituent-task,-16.166151
...,...,...,...,...,...,...,...,...,...,...
93,passive,Qwen/Qwen2.5-1.5B-Instruct,0.364126,semantics,passive,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.431192,semantics,passive,18.418459
105,sherlock-negation,Qwen/Qwen2.5-1.5B-Instruct,0.006695,reasoning,negation,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.008264,reasoning,negation,23.450414
6,blimp-animate_subject_trans,Qwen/Qwen2.5-1.5B-Instruct,0.722826,syntax,argument structure,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.895092,syntax,argument structure,23.832410
17,blimp-determiner_noun_agreement_with_adj_irreg...,Qwen/Qwen2.5-1.5B-Instruct,0.441847,morphology,determiner noun agreement,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.565224,morphology,determiner noun agreement,27.922964


In [513]:
all_probes_3b = calculate_score_change(
    final_set=final_set_sorted,
    model_name_x='Qwen/Qwen2.5-3B-Instruct', 
    model_name_y='Qwen/Qwen2.5-3B-Instruct-AWQ',
    column_group='probe'
)
all_probes_3b

,probe,model_name_x,score_x,linguistic subfield_x,linguistic phenomena_x,model_name_y,score_y,linguistic subfield_y,linguistic phenomena_y,score_change_pct
103,senteval-word_content,Qwen/Qwen2.5-3B-Instruct,0.003246,semantics,word content,Qwen/Qwen2.5-3B-Instruct-AWQ,0.002456,semantics,word content,-24.342382
124,zorro-local_attractor-in_question_with_aux,Qwen/Qwen2.5-3B-Instruct,0.755839,syntax,local attractor,Qwen/Qwen2.5-3B-Instruct-AWQ,0.613958,syntax,local attractor,-18.771335
92,olmpics-size_comparison,Qwen/Qwen2.5-3B-Instruct,0.625038,reasoning,object comparision,Qwen/Qwen2.5-3B-Instruct-AWQ,0.531119,reasoning,object comparision,-15.026104
110,zorro-agreement_subject_verb-across_relative_c...,Qwen/Qwen2.5-3B-Instruct,0.686333,syntax,subject-verb agreement,Qwen/Qwen2.5-3B-Instruct-AWQ,0.625583,syntax,subject-verb agreement,-8.851362
15,blimp-determiner_noun_agreement_with_adj_2,Qwen/Qwen2.5-3B-Instruct,0.568532,morphology,determiner noun agreement,Qwen/Qwen2.5-3B-Instruct-AWQ,0.524891,morphology,determiner noun agreement,-7.676082
...,...,...,...,...,...,...,...,...,...,...
109,zorro-agreement_subject_verb-across_prepositio...,Qwen/Qwen2.5-3B-Instruct,0.468128,syntax,subject-verb agreement,Qwen/Qwen2.5-3B-Instruct-AWQ,0.507612,syntax,subject-verb agreement,8.434502
123,zorro-island-effects-coordinate_structure_cons...,Qwen/Qwen2.5-3B-Instruct,0.729940,syntax,island effects,Qwen/Qwen2.5-3B-Instruct-AWQ,0.805952,syntax,island effects,10.413333
93,passive,Qwen/Qwen2.5-3B-Instruct,0.389100,semantics,passive,Qwen/Qwen2.5-3B-Instruct-AWQ,0.449814,semantics,passive,15.603577
119,zorro-ellipsis-n_bar,Qwen/Qwen2.5-3B-Instruct,0.670826,syntax,ellipsis,Qwen/Qwen2.5-3B-Instruct-AWQ,0.807039,syntax,ellipsis,20.305268


In [514]:
all_probes_7b = calculate_score_change(
    final_set=final_set_sorted,
    model_name_x='Qwen/Qwen2.5-7B-Instruct', 
    model_name_y='Qwen/Qwen2.5-7B-Instruct-AWQ',
    column_group='probe'
)
all_probes_7b

,probe,model_name_x,score_x,linguistic subfield_x,linguistic phenomena_x,model_name_y,score_y,linguistic subfield_y,linguistic phenomena_y,score_change_pct
84,next-sentence-prediction,Qwen/Qwen2.5-7B-Instruct,0.000399,discourse,next sentence prediction,Qwen/Qwen2.5-7B-Instruct-AWQ,0.000000,discourse,next sentence prediction,-100.000000
105,sherlock-negation,Qwen/Qwen2.5-7B-Instruct,0.009677,reasoning,negation,Qwen/Qwen2.5-7B-Instruct-AWQ,0.000000,reasoning,negation,-100.000000
44,blimp-principle_A_c_command,Qwen/Qwen2.5-7B-Instruct,0.407293,syntax,binding,Qwen/Qwen2.5-7B-Instruct-AWQ,0.256417,syntax,binding,-37.043647
121,zorro-filler-gap-wh_question_subject,Qwen/Qwen2.5-7B-Instruct,0.957304,syntax,filler gap,Qwen/Qwen2.5-7B-Instruct-AWQ,0.755459,syntax,filler gap,-21.084709
35,blimp-left_branch_island_echo_question,Qwen/Qwen2.5-7B-Instruct,0.982911,syntax,island effects,Qwen/Qwen2.5-7B-Instruct-AWQ,0.795531,syntax,island effects,-19.063763
...,...,...,...,...,...,...,...,...,...,...
124,zorro-local_attractor-in_question_with_aux,Qwen/Qwen2.5-7B-Instruct,0.544415,syntax,local attractor,Qwen/Qwen2.5-7B-Instruct-AWQ,0.796515,syntax,local attractor,46.306686
34,blimp-irregular_plural_subject_verb_agreement_2,Qwen/Qwen2.5-7B-Instruct,0.531009,morphology,subject-verb agreement,Qwen/Qwen2.5-7B-Instruct-AWQ,0.822973,morphology,subject-verb agreement,54.982915
55,blimp-sentential_subject_island,Qwen/Qwen2.5-7B-Instruct,0.385024,syntax,island effects,Qwen/Qwen2.5-7B-Instruct-AWQ,0.638227,syntax,island effects,65.762851
111,zorro-agreement_subject_verb-in_question_with_aux,Qwen/Qwen2.5-7B-Instruct,0.203670,syntax,subject-verb agreement,Qwen/Qwen2.5-7B-Instruct-AWQ,0.602062,syntax,subject-verb agreement,195.607042


# Over Phenomena

In [515]:
all_phenomena_05b = calculate_score_change(
    final_set=final_set_phenomena_sorted, 
    model_name_x='Qwen/Qwen2.5-0.5B-Instruct', 
    model_name_y='Qwen/Qwen2.5-0.5B-Instruct-AWQ'
)
all_phenomena_05b

,linguistic phenomena,model_name_x,score_x,model_name_y,score_y,score_change_pct
29,passive,Qwen/Qwen2.5-0.5B-Instruct,0.172720,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.097753,-43.403875
47,word content,Qwen/Qwen2.5-0.5B-Instruct,0.000374,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.000213,-42.878630
27,object gender,Qwen/Qwen2.5-0.5B-Instruct,0.475838,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.333665,-29.878414
14,discourse connective,Qwen/Qwen2.5-0.5B-Instruct,0.069996,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.053380,-23.738686
16,encyclopedic composition,Qwen/Qwen2.5-0.5B-Instruct,0.792372,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.640000,-19.229806
33,semantic odd man out,Qwen/Qwen2.5-0.5B-Instruct,0.570771,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.478975,-16.082793
41,synonym-/antonym-detection,Qwen/Qwen2.5-0.5B-Instruct,0.355676,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.311352,-12.462005
2,anaphor agreement,Qwen/Qwen2.5-0.5B-Instruct,0.437465,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.393409,-10.070778
17,filler gap,Qwen/Qwen2.5-0.5B-Instruct,0.707497,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.643422,-9.056579
23,negative polarity item licensing,Qwen/Qwen2.5-0.5B-Instruct,0.700267,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.659865,-5.769412


In [516]:
all_phenomena_15b  = calculate_score_change(
    final_set=final_set_phenomena_sorted, 
    model_name_x='Qwen/Qwen2.5-1.5B-Instruct', 
    model_name_y='Qwen/Qwen2.5-1.5B-Instruct-AWQ'
)
all_phenomena_15b 

,linguistic phenomena,model_name_x,score_x,model_name_y,score_y,score_change_pct
47,word content,Qwen/Qwen2.5-1.5B-Instruct,0.000846,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.000366,-56.670974
41,synonym-/antonym-detection,Qwen/Qwen2.5-1.5B-Instruct,0.314290,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.233311,-25.765701
14,discourse connective,Qwen/Qwen2.5-1.5B-Instruct,0.073738,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.060835,-17.498662
26,object comparision,Qwen/Qwen2.5-1.5B-Instruct,0.487882,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.407964,-16.380633
44,top-constituent-task,Qwen/Qwen2.5-1.5B-Instruct,0.445716,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.373661,-16.166151
0,age comparison,Qwen/Qwen2.5-1.5B-Instruct,0.425719,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.402041,-5.561928
12,deoncausative-inchoative alternation,Qwen/Qwen2.5-1.5B-Instruct,0.782686,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.752083,-3.909962
45,tree-depth,Qwen/Qwen2.5-1.5B-Instruct,0.214844,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.210245,-2.140840
33,semantic odd man out,Qwen/Qwen2.5-1.5B-Instruct,0.618743,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.610892,-1.268867
28,object number,Qwen/Qwen2.5-1.5B-Instruct,0.773647,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.764292,-1.209263


In [517]:
all_phenomena_3b  = calculate_score_change(
    final_set=final_set_phenomena_sorted, 
    model_name_x='Qwen/Qwen2.5-3B-Instruct', 
    model_name_y='Qwen/Qwen2.5-3B-Instruct-AWQ'
)
all_phenomena_3b 

,linguistic phenomena,model_name_x,score_x,model_name_y,score_y,score_change_pct
47,word content,Qwen/Qwen2.5-3B-Instruct,0.003246,Qwen/Qwen2.5-3B-Instruct-AWQ,0.002456,-24.342382
20,local attractor,Qwen/Qwen2.5-3B-Instruct,0.755839,Qwen/Qwen2.5-3B-Instruct-AWQ,0.613958,-18.771335
26,object comparision,Qwen/Qwen2.5-3B-Instruct,0.625038,Qwen/Qwen2.5-3B-Instruct-AWQ,0.531119,-15.026104
14,discourse connective,Qwen/Qwen2.5-3B-Instruct,0.082180,Qwen/Qwen2.5-3B-Instruct-AWQ,0.076780,-6.571542
38,subject gender,Qwen/Qwen2.5-3B-Instruct,0.854115,Qwen/Qwen2.5-3B-Instruct-AWQ,0.798406,-6.522347
13,determiner noun agreement,Qwen/Qwen2.5-3B-Instruct,0.580737,Qwen/Qwen2.5-3B-Instruct-AWQ,0.559867,-3.593726
0,age comparison,Qwen/Qwen2.5-3B-Instruct,0.354422,Qwen/Qwen2.5-3B-Instruct-AWQ,0.346667,-2.188099
28,object number,Qwen/Qwen2.5-3B-Instruct,0.751516,Qwen/Qwen2.5-3B-Instruct-AWQ,0.738637,-1.713762
40,subject-verb agreement,Qwen/Qwen2.5-3B-Instruct,0.687026,Qwen/Qwen2.5-3B-Instruct-AWQ,0.678384,-1.257869
31,quantifiers,Qwen/Qwen2.5-3B-Instruct,0.971562,Qwen/Qwen2.5-3B-Instruct-AWQ,0.960953,-1.092027


In [518]:
all_phenomena_7b  = calculate_score_change(
    final_set=final_set_phenomena_sorted, 
    model_name_x='Qwen/Qwen2.5-7B-Instruct', 
    model_name_y='Qwen/Qwen2.5-7B-Instruct-AWQ'
)
all_phenomena_7b

,linguistic phenomena,model_name_x,score_x,model_name_y,score_y,score_change_pct
24,next sentence prediction,Qwen/Qwen2.5-7B-Instruct,0.000399,Qwen/Qwen2.5-7B-Instruct-AWQ,0.000000,-100.000000
35,sentiment analysis,Qwen/Qwen2.5-7B-Instruct,0.200037,Qwen/Qwen2.5-7B-Instruct-AWQ,0.162513,-18.758683
44,top-constituent-task,Qwen/Qwen2.5-7B-Instruct,0.494533,Qwen/Qwen2.5-7B-Instruct-AWQ,0.441795,-10.664281
3,antonym negation,Qwen/Qwen2.5-7B-Instruct,0.447825,Qwen/Qwen2.5-7B-Instruct-AWQ,0.412350,-7.921615
27,object gender,Qwen/Qwen2.5-7B-Instruct,0.475008,Qwen/Qwen2.5-7B-Instruct-AWQ,0.465226,-2.059297
22,negation,Qwen/Qwen2.5-7B-Instruct,0.213842,Qwen/Qwen2.5-7B-Instruct-AWQ,0.211361,-1.159842
33,semantic odd man out,Qwen/Qwen2.5-7B-Instruct,0.639568,Qwen/Qwen2.5-7B-Instruct-AWQ,0.632373,-1.124987
25,object animacy,Qwen/Qwen2.5-7B-Instruct,0.924855,Qwen/Qwen2.5-7B-Instruct-AWQ,0.915671,-0.993039
6,binding,Qwen/Qwen2.5-7B-Instruct,0.646229,Qwen/Qwen2.5-7B-Instruct-AWQ,0.643266,-0.458438
36,speculation,Qwen/Qwen2.5-7B-Instruct,0.399546,Qwen/Qwen2.5-7B-Instruct-AWQ,0.399093,-0.113488


# Over all Subfields

In [519]:
all_subfields_05b = calculate_score_change(
    final_set=final_set_subfield_sorted, 
    model_name_x='Qwen/Qwen2.5-0.5B-Instruct', 
    model_name_y='Qwen/Qwen2.5-0.5B-Instruct-AWQ',
    column_group='linguistic subfield'
)
all_subfields_05b

,linguistic subfield,model_name_x,score_x,model_name_y,score_y,score_change_pct
0,discourse,Qwen/Qwen2.5-0.5B-Instruct,0.289935,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.284397,-1.910326
4,syntax,Qwen/Qwen2.5-0.5B-Instruct,0.600917,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.593928,-1.163032
3,semantics,Qwen/Qwen2.5-0.5B-Instruct,0.600007,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.598146,-0.310170
1,morphology,Qwen/Qwen2.5-0.5B-Instruct,0.469297,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.483613,3.050567
2,reasoning,Qwen/Qwen2.5-0.5B-Instruct,0.347619,Qwen/Qwen2.5-0.5B-Instruct-AWQ,0.366886,5.542375


In [520]:
all_subfields_15b = calculate_score_change(
    final_set=final_set_subfield_sorted, 
    model_name_x='Qwen/Qwen2.5-1.5B-Instruct', 
    model_name_y='Qwen/Qwen2.5-1.5B-Instruct-AWQ',
    column_group='linguistic subfield'
)
all_subfields_15b

,linguistic subfield,model_name_x,score_x,model_name_y,score_y,score_change_pct
0,discourse,Qwen/Qwen2.5-1.5B-Instruct,0.293605,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.289310,-1.462836
2,reasoning,Qwen/Qwen2.5-1.5B-Instruct,0.385483,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.380550,-1.279694
3,semantics,Qwen/Qwen2.5-1.5B-Instruct,0.762766,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.764153,0.181855
4,syntax,Qwen/Qwen2.5-1.5B-Instruct,0.751931,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.761171,1.228730
1,morphology,Qwen/Qwen2.5-1.5B-Instruct,0.652050,Qwen/Qwen2.5-1.5B-Instruct-AWQ,0.661183,1.400601


In [521]:
all_subfields_3b = calculate_score_change(
    final_set=final_set_subfield_sorted, 
    model_name_x='Qwen/Qwen2.5-3B-Instruct', 
    model_name_y='Qwen/Qwen2.5-3B-Instruct-AWQ',
    column_group='linguistic subfield'
)
all_subfields_3b

,linguistic subfield,model_name_x,score_x,model_name_y,score_y,score_change_pct
2,reasoning,Qwen/Qwen2.5-3B-Instruct,0.392680,Qwen/Qwen2.5-3B-Instruct-AWQ,0.384639,-2.047865
1,morphology,Qwen/Qwen2.5-3B-Instruct,0.679330,Qwen/Qwen2.5-3B-Instruct-AWQ,0.669172,-1.495329
0,discourse,Qwen/Qwen2.5-3B-Instruct,0.293917,Qwen/Qwen2.5-3B-Instruct-AWQ,0.290815,-1.055560
4,syntax,Qwen/Qwen2.5-3B-Instruct,0.771880,Qwen/Qwen2.5-3B-Instruct-AWQ,0.773295,0.183300
3,semantics,Qwen/Qwen2.5-3B-Instruct,0.765313,Qwen/Qwen2.5-3B-Instruct-AWQ,0.770589,0.689396


In [522]:
all_subfields_7b = calculate_score_change(
    final_set=final_set_subfield_sorted, 
    model_name_x='Qwen/Qwen2.5-7B-Instruct', 
    model_name_y='Qwen/Qwen2.5-7B-Instruct-AWQ',
    column_group='linguistic subfield'
)
all_subfields_7b

,linguistic subfield,model_name_x,score_x,model_name_y,score_y,score_change_pct
0,discourse,Qwen/Qwen2.5-7B-Instruct,0.296533,Qwen/Qwen2.5-7B-Instruct-AWQ,0.297879,0.454022
2,reasoning,Qwen/Qwen2.5-7B-Instruct,0.394401,Qwen/Qwen2.5-7B-Instruct-AWQ,0.400313,1.498831
4,syntax,Qwen/Qwen2.5-7B-Instruct,0.718948,Qwen/Qwen2.5-7B-Instruct-AWQ,0.750170,4.342730
3,semantics,Qwen/Qwen2.5-7B-Instruct,0.705434,Qwen/Qwen2.5-7B-Instruct-AWQ,0.753611,6.829456
1,morphology,Qwen/Qwen2.5-7B-Instruct,0.614720,Qwen/Qwen2.5-7B-Instruct-AWQ,0.666907,8.489603


# Different Quantizations

In [523]:
qwen_results_halfp = qwen_results_halfp.merge(
    mapping_subfield,
    how='left',
    left_on='probing_dataset',
    right_on='probing dataset'
).merge(
    mapping_phenomena,
    how='left',
    left_on='probing_dataset',
    right_on='probing dataset'
).rename(columns={'probing_dataset': 'probe'}).drop(columns=['Unnamed: 0', 'probing dataset_x', 'probing dataset_y'])
qwen_results_halfp['model_name'] = qwen_results_halfp['model_name'].str.replace('__', '/')
qwen_results_halfp['model_name'] = qwen_results_halfp['model_name'].apply(lambda x: x + "-HalfP")
qwen_results_halfp

,probe,model_name,score,linguistic subfield,linguistic phenomena
0,SemAntoNeg,Qwen/Qwen2.5-0.5B-Instruct-HalfP,0.000000,reasoning,negation
1,SemAntoNeg,Qwen/Qwen2.5-0.5B-Instruct-AWQ-HalfP,0.000000,reasoning,negation
2,SemAntoNeg,Qwen/Qwen2.5-1.5B-Instruct-HalfP,0.000000,reasoning,negation
3,SemAntoNeg,Qwen/Qwen2.5-1.5B-Instruct-AWQ-HalfP,0.000000,reasoning,negation
4,SemAntoNeg,Qwen/Qwen2.5-3B-Instruct-HalfP,0.000000,reasoning,negation
...,...,...,...,...,...
898,zorro-quantifiers-superlative,Qwen/Qwen2.5-1.5B-Instruct-HalfP,0.948039,semantics,quantifiers
899,zorro-quantifiers-superlative,Qwen/Qwen2.5-1.5B-Instruct-AWQ-HalfP,0.928130,semantics,quantifiers
900,zorro-quantifiers-superlative,Qwen/Qwen2.5-3B-Instruct-HalfP,0.968133,semantics,quantifiers
901,zorro-quantifiers-superlative,Qwen/Qwen2.5-3B-Instruct-AWQ-HalfP,0.938919,semantics,quantifiers


In [524]:
qwen_results_8bit = qwen_results_8bit.merge(
    mapping_subfield,
    how='left',
    left_on='probing_dataset',
    right_on='probing dataset'
).merge(
    mapping_phenomena,
    how='left',
    left_on='probing_dataset',
    right_on='probing dataset'
).rename(columns={'probing_dataset': 'probe'}).drop(columns=['Unnamed: 0', 'probing dataset_x', 'probing dataset_y'])
qwen_results_8bit['model_name'] = qwen_results_8bit['model_name'].str.replace('__', '/')
qwen_results_8bit['model_name'] = qwen_results_8bit['model_name'].apply(lambda x: x + "-8Bit")
qwen_results_8bit

,probe,model_name,score,linguistic subfield,linguistic phenomena
0,SemAntoNeg,Qwen/Qwen2.5-0.5B-Instruct-8Bit,0.000000,reasoning,negation
1,SemAntoNeg,Qwen/Qwen2.5-1.5B-Instruct-8Bit,0.000000,reasoning,negation
2,SemAntoNeg,Qwen/Qwen2.5-3B-Instruct-8Bit,0.000000,reasoning,negation
3,SemAntoNeg,Qwen/Qwen2.5-7B-Instruct-8Bit,0.000000,reasoning,negation
4,bioscope-org-negation,Qwen/Qwen2.5-0.5B-Instruct-8Bit,0.348465,reasoning,negation
...,...,...,...,...,...
511,zorro-quantifiers-existential_there,Qwen/Qwen2.5-7B-Instruct-8Bit,0.837179,semantics,quantifiers
512,zorro-quantifiers-superlative,Qwen/Qwen2.5-0.5B-Instruct-8Bit,0.766732,semantics,quantifiers
513,zorro-quantifiers-superlative,Qwen/Qwen2.5-1.5B-Instruct-8Bit,0.966251,semantics,quantifiers
514,zorro-quantifiers-superlative,Qwen/Qwen2.5-3B-Instruct-8Bit,0.934165,semantics,quantifiers


In [525]:
qwen_results_4bit = qwen_results_4bit.merge(
    mapping_subfield,
    how='left',
    left_on='probing_dataset',
    right_on='probing dataset'
).merge(
    mapping_phenomena,
    how='left',
    left_on='probing_dataset',
    right_on='probing dataset'
).rename(columns={'probing_dataset': 'probe'}).drop(columns=['Unnamed: 0', 'probing dataset_x', 'probing dataset_y'])
qwen_results_4bit['model_name'] = qwen_results_4bit['model_name'].str.replace('__', '/')
qwen_results_4bit['model_name'] = qwen_results_4bit['model_name'].apply(lambda x: x + "-4Bit")
qwen_results_4bit

,probe,model_name,score,linguistic subfield,linguistic phenomena
0,SemAntoNeg,Qwen/Qwen2.5-0.5B-Instruct-4Bit,0.000000,reasoning,negation
1,SemAntoNeg,Qwen/Qwen2.5-1.5B-Instruct-4Bit,0.000000,reasoning,negation
2,SemAntoNeg,Qwen/Qwen2.5-3B-Instruct-4Bit,0.000000,reasoning,negation
3,SemAntoNeg,Qwen/Qwen2.5-7B-Instruct-4Bit,0.000000,reasoning,negation
4,bioscope-org-negation,Qwen/Qwen2.5-0.5B-Instruct-4Bit,0.354660,reasoning,negation
...,...,...,...,...,...
511,zorro-quantifiers-existential_there,Qwen/Qwen2.5-7B-Instruct-4Bit,0.851654,semantics,quantifiers
512,zorro-quantifiers-superlative,Qwen/Qwen2.5-0.5B-Instruct-4Bit,0.786813,semantics,quantifiers
513,zorro-quantifiers-superlative,Qwen/Qwen2.5-1.5B-Instruct-4Bit,0.967719,semantics,quantifiers
514,zorro-quantifiers-superlative,Qwen/Qwen2.5-3B-Instruct-4Bit,0.791649,semantics,quantifiers


# Qwen Results HalfP

In [526]:
HalfP_05b = qwen_results_halfp[qwen_results_halfp['model_name'].str.contains('Qwen/Qwen2.5-0.5B-Instruct-HalfP')]
halfp05concat = pd.concat([HalfP_05b, final_set_sorted])
all_probes_halfp05 = calculate_score_change(
    final_set=halfp05concat,
    model_name_x='Qwen/Qwen2.5-0.5B-Instruct-HalfP', 
    model_name_y='Qwen/Qwen2.5-0.5B-Instruct',
    column_group='probe'
)
all_probes_halfp05

,probe,model_name_x,score_x,linguistic subfield_x,linguistic phenomena_x,model_name_y,score_y,linguistic subfield_y,linguistic phenomena_y,score_change_pct
89,olmpics-conjunction_filt4,Qwen/Qwen2.5-0.5B-Instruct-HalfP,0.007407,reasoning,property conjunction,Qwen/Qwen2.5-0.5B-Instruct,0.000000,reasoning,property conjunction,-100.000000
47,blimp-principle_A_domain_1,Qwen/Qwen2.5-0.5B-Instruct-HalfP,0.461483,syntax,binding,Qwen/Qwen2.5-0.5B-Instruct,0.316521,syntax,binding,-31.412173
93,passive,Qwen/Qwen2.5-0.5B-Instruct-HalfP,0.240709,semantics,passive,Qwen/Qwen2.5-0.5B-Instruct,0.172720,semantics,passive,-28.245493
92,olmpics-size_comparison,Qwen/Qwen2.5-0.5B-Instruct-HalfP,0.282373,reasoning,object comparision,Qwen/Qwen2.5-0.5B-Instruct,0.204155,reasoning,object comparision,-27.700259
25,blimp-existential_there_quantifiers_1,Qwen/Qwen2.5-0.5B-Instruct-HalfP,0.677954,semantics,quantifiers,Qwen/Qwen2.5-0.5B-Instruct,0.502345,semantics,quantifiers,-25.902817
...,...,...,...,...,...,...,...,...,...,...
20,blimp-distractor_agreement_relative_clause,Qwen/Qwen2.5-0.5B-Instruct-HalfP,0.182870,morphology,subject-verb agreement,Qwen/Qwen2.5-0.5B-Instruct,0.336645,morphology,subject-verb agreement,84.089422
127,zorro-quantifiers-existential_there,Qwen/Qwen2.5-0.5B-Instruct-HalfP,0.416844,semantics,quantifiers,Qwen/Qwen2.5-0.5B-Instruct,0.775349,semantics,quantifiers,86.004633
110,zorro-agreement_subject_verb-across_relative_c...,Qwen/Qwen2.5-0.5B-Instruct-HalfP,0.296583,syntax,subject-verb agreement,Qwen/Qwen2.5-0.5B-Instruct,0.579879,syntax,subject-verb agreement,95.520324
109,zorro-agreement_subject_verb-across_prepositio...,Qwen/Qwen2.5-0.5B-Instruct-HalfP,0.290851,syntax,subject-verb agreement,Qwen/Qwen2.5-0.5B-Instruct,0.576943,syntax,subject-verb agreement,98.363910


In [527]:
HalfP_15b = qwen_results_halfp[qwen_results_halfp['model_name'].str.contains('Qwen/Qwen2.5-1.5B-Instruct-HalfP')]
halfp15concat = pd.concat([HalfP_15b, final_set_sorted])
all_probes_halfp15 = calculate_score_change(
    final_set=halfp15concat,
    model_name_x='Qwen/Qwen2.5-1.5B-Instruct-HalfP', 
    model_name_y='Qwen/Qwen2.5-1.5B-Instruct',
    column_group='probe'
)
all_probes_halfp15

,probe,model_name_x,score_x,linguistic subfield_x,linguistic phenomena_x,model_name_y,score_y,linguistic subfield_y,linguistic phenomena_y,score_change_pct
105,sherlock-negation,Qwen/Qwen2.5-1.5B-Instruct-HalfP,0.012955,reasoning,negation,Qwen/Qwen2.5-1.5B-Instruct,0.006695,reasoning,negation,-48.326362
103,senteval-word_content,Qwen/Qwen2.5-1.5B-Instruct-HalfP,0.001517,semantics,word content,Qwen/Qwen2.5-1.5B-Instruct,0.000846,semantics,word content,-44.264909
107,synonym-antonym-hard,Qwen/Qwen2.5-1.5B-Instruct-HalfP,0.472649,semantics,synonym-/antonym-detection,Qwen/Qwen2.5-1.5B-Instruct,0.314290,semantics,synonym-/antonym-detection,-33.504572
85,olmpics-antonym_synonym_negation,Qwen/Qwen2.5-1.5B-Instruct-HalfP,0.615225,reasoning,antonym negation,Qwen/Qwen2.5-1.5B-Instruct,0.443164,reasoning,antonym negation,-27.967147
6,blimp-animate_subject_trans,Qwen/Qwen2.5-1.5B-Instruct-HalfP,0.906466,syntax,argument structure,Qwen/Qwen2.5-1.5B-Instruct,0.722826,syntax,argument structure,-20.258933
...,...,...,...,...,...,...,...,...,...,...
38,blimp-npi_present_1,Qwen/Qwen2.5-1.5B-Instruct-HalfP,0.757497,semantics,negative polarity item licensing,Qwen/Qwen2.5-1.5B-Instruct,0.947056,semantics,negative polarity item licensing,25.024516
43,blimp-passive_2,Qwen/Qwen2.5-1.5B-Instruct-HalfP,0.612270,syntax,argument structure,Qwen/Qwen2.5-1.5B-Instruct,0.773747,syntax,argument structure,26.373638
68,blimp-wh_vs_that_with_gap_long_distance,Qwen/Qwen2.5-1.5B-Instruct-HalfP,0.687673,syntax,filler gap,Qwen/Qwen2.5-1.5B-Instruct,0.870746,syntax,filler gap,26.622057
112,zorro-agreement_subject_verb-in_simple_question,Qwen/Qwen2.5-1.5B-Instruct-HalfP,0.406021,syntax,subject-verb agreement,Qwen/Qwen2.5-1.5B-Instruct,0.524139,syntax,subject-verb agreement,29.091764


In [528]:
HalfP_3b = qwen_results_halfp[qwen_results_halfp['model_name'].str.contains('Qwen/Qwen2.5-3B-Instruct-HalfP')]
halfp3concat = pd.concat([HalfP_3b, final_set_sorted])
all_probes_halfp3 = calculate_score_change(
    final_set=halfp3concat,
    model_name_x='Qwen/Qwen2.5-3B-Instruct-HalfP', 
    model_name_y='Qwen/Qwen2.5-3B-Instruct',
    column_group='probe'
)
all_probes_halfp3

,probe,model_name_x,score_x,linguistic subfield_x,linguistic phenomena_x,model_name_y,score_y,linguistic subfield_y,linguistic phenomena_y,score_change_pct
84,next-sentence-prediction,Qwen/Qwen2.5-3B-Instruct-HalfP,0.000798,discourse,next sentence prediction,Qwen/Qwen2.5-3B-Instruct,0.000000,discourse,next sentence prediction,-100.000000
91,olmpics-number_comparison_age_compare_masked,Qwen/Qwen2.5-3B-Instruct-HalfP,0.534535,reasoning,age comparison,Qwen/Qwen2.5-3B-Instruct,0.354422,reasoning,age comparison,-33.695267
107,synonym-antonym-hard,Qwen/Qwen2.5-3B-Instruct-HalfP,0.280538,semantics,synonym-/antonym-detection,Qwen/Qwen2.5-3B-Instruct,0.195705,semantics,synonym-/antonym-detection,-30.239380
85,olmpics-antonym_synonym_negation,Qwen/Qwen2.5-3B-Instruct-HalfP,0.627829,reasoning,antonym negation,Qwen/Qwen2.5-3B-Instruct,0.480309,reasoning,antonym negation,-23.496810
109,zorro-agreement_subject_verb-across_prepositio...,Qwen/Qwen2.5-3B-Instruct-HalfP,0.556388,syntax,subject-verb agreement,Qwen/Qwen2.5-3B-Instruct,0.468128,syntax,subject-verb agreement,-15.863012
...,...,...,...,...,...,...,...,...,...,...
82,flesch,Qwen/Qwen2.5-3B-Instruct-HalfP,0.219037,syntax,readability,Qwen/Qwen2.5-3B-Instruct,0.245491,syntax,readability,12.077573
93,passive,Qwen/Qwen2.5-3B-Instruct-HalfP,0.337367,semantics,passive,Qwen/Qwen2.5-3B-Instruct,0.389100,semantics,passive,15.334223
73,context-object_gender,Qwen/Qwen2.5-3B-Instruct-HalfP,0.400564,semantics,object gender,Qwen/Qwen2.5-3B-Instruct,0.482259,semantics,object gender,20.394913
55,blimp-sentential_subject_island,Qwen/Qwen2.5-3B-Instruct-HalfP,0.480414,syntax,island effects,Qwen/Qwen2.5-3B-Instruct,0.584342,syntax,island effects,21.632937


In [529]:
HalfP_7b = qwen_results_halfp[qwen_results_halfp['model_name'].str.contains('Qwen/Qwen2.5-7B-Instruct-HalfP')]
halfp7concat = pd.concat([HalfP_7b, final_set_sorted])
all_probes_halfp7 = calculate_score_change(
    final_set=halfp7concat,
    model_name_x='Qwen/Qwen2.5-7B-Instruct-HalfP', 
    model_name_y='Qwen/Qwen2.5-7B-Instruct',
    column_group='probe'
)
all_probes_halfp7

,probe,model_name_x,score_x,linguistic subfield_x,linguistic phenomena_x,model_name_y,score_y,linguistic subfield_y,linguistic phenomena_y,score_change_pct
89,olmpics-conjunction_filt4,Qwen/Qwen2.5-7B-Instruct-HalfP,0.007407,reasoning,property conjunction,Qwen/Qwen2.5-7B-Instruct,0.000000,reasoning,property conjunction,-100.000000
107,synonym-antonym-hard,Qwen/Qwen2.5-7B-Instruct-HalfP,0.227835,semantics,synonym-/antonym-detection,Qwen/Qwen2.5-7B-Instruct,0.038782,semantics,synonym-/antonym-detection,-82.978102
105,sherlock-negation,Qwen/Qwen2.5-7B-Instruct-HalfP,0.046388,reasoning,negation,Qwen/Qwen2.5-7B-Instruct,0.009677,reasoning,negation,-79.138216
108,zorro-agreement_determiner_noun-between_neighbors,Qwen/Qwen2.5-7B-Instruct-HalfP,0.434198,morphology,determiner noun agreement,Qwen/Qwen2.5-7B-Instruct,0.277665,morphology,determiner noun agreement,-36.050964
111,zorro-agreement_subject_verb-in_question_with_aux,Qwen/Qwen2.5-7B-Instruct-HalfP,0.294979,syntax,subject-verb agreement,Qwen/Qwen2.5-7B-Instruct,0.203670,syntax,subject-verb agreement,-30.954473
...,...,...,...,...,...,...,...,...,...,...
22,blimp-ellipsis_n_bar_1,Qwen/Qwen2.5-7B-Instruct-HalfP,0.432181,syntax,ellipsis,Qwen/Qwen2.5-7B-Instruct,0.668113,syntax,ellipsis,54.591061
33,blimp-irregular_plural_subject_verb_agreement_1,Qwen/Qwen2.5-7B-Instruct-HalfP,0.439913,morphology,subject-verb agreement,Qwen/Qwen2.5-7B-Instruct,0.758226,morphology,subject-verb agreement,72.357984
91,olmpics-number_comparison_age_compare_masked,Qwen/Qwen2.5-7B-Instruct-HalfP,0.296296,reasoning,age comparison,Qwen/Qwen2.5-7B-Instruct,0.616232,reasoning,age comparison,107.978194
48,blimp-principle_A_domain_2,Qwen/Qwen2.5-7B-Instruct-HalfP,0.177347,syntax,binding,Qwen/Qwen2.5-7B-Instruct,0.427550,syntax,binding,141.080716


# Qwen Results 8Bit

In [530]:
EightBit_05b = qwen_results_8bit[qwen_results_8bit['model_name'].str.contains('Qwen/Qwen2.5-0.5B-Instruct-8Bit')]
eightbit05concat = pd.concat([EightBit_05b, final_set_sorted])
all_probes_8bit05 = calculate_score_change(
    final_set=eightbit05concat,
    model_name_x='Qwen/Qwen2.5-0.5B-Instruct-8Bit', 
    model_name_y='Qwen/Qwen2.5-0.5B-Instruct',
    column_group='probe'
)
all_probes_8bit05

,probe,model_name_x,score_x,linguistic subfield_x,linguistic phenomena_x,model_name_y,score_y,linguistic subfield_y,linguistic phenomena_y,score_change_pct
103,senteval-word_content,Qwen/Qwen2.5-0.5B-Instruct-8Bit,0.000536,semantics,word content,Qwen/Qwen2.5-0.5B-Instruct,0.000374,semantics,word content,-30.345815
108,zorro-agreement_determiner_noun-between_neighbors,Qwen/Qwen2.5-0.5B-Instruct-8Bit,0.565885,morphology,determiner noun agreement,Qwen/Qwen2.5-0.5B-Instruct,0.404919,morphology,determiner noun agreement,-28.445063
92,olmpics-size_comparison,Qwen/Qwen2.5-0.5B-Instruct-8Bit,0.285118,reasoning,object comparision,Qwen/Qwen2.5-0.5B-Instruct,0.204155,reasoning,object comparision,-28.396273
52,blimp-regular_plural_subject_verb_agreement_2,Qwen/Qwen2.5-0.5B-Instruct-8Bit,0.713873,morphology,subject-verb agreement,Qwen/Qwen2.5-0.5B-Instruct,0.519663,morphology,subject-verb agreement,-27.205077
29,blimp-inchoative,Qwen/Qwen2.5-0.5B-Instruct-8Bit,0.619892,syntax,argument structure,Qwen/Qwen2.5-0.5B-Instruct,0.455289,syntax,argument structure,-26.553456
...,...,...,...,...,...,...,...,...,...,...
122,zorro-island-effects-adjunct_island,Qwen/Qwen2.5-0.5B-Instruct-8Bit,0.420755,syntax,island effects,Qwen/Qwen2.5-0.5B-Instruct,0.714677,syntax,island effects,69.855752
124,zorro-local_attractor-in_question_with_aux,Qwen/Qwen2.5-0.5B-Instruct-8Bit,0.247839,syntax,local attractor,Qwen/Qwen2.5-0.5B-Instruct,0.421284,syntax,local attractor,69.982730
17,blimp-determiner_noun_agreement_with_adj_irreg...,Qwen/Qwen2.5-0.5B-Instruct-8Bit,0.328749,morphology,determiner noun agreement,Qwen/Qwen2.5-0.5B-Instruct,0.581147,morphology,determiner noun agreement,76.775094
110,zorro-agreement_subject_verb-across_relative_c...,Qwen/Qwen2.5-0.5B-Instruct-8Bit,0.263265,syntax,subject-verb agreement,Qwen/Qwen2.5-0.5B-Instruct,0.579879,syntax,subject-verb agreement,120.264469


# Qwen Results 4Bit

In [531]:
mapping_bruh = other_models_og[['probing dataset', 'linguistic phenomena', 'linguistic subfield']].drop_duplicates(subset=['probing dataset']).sort_values(by=['linguistic subfield', 'linguistic phenomena', 'probing dataset'], ascending=[True, True, True])
mapping_bruh = mapping_bruh[['linguistic subfield', 'linguistic phenomena', 'probing dataset']].set_index(['linguistic subfield', 'linguistic phenomena'])

In [532]:
latex_context = mapping_bruh.to_latex(escape=True, sparsify=True, multirow=True, multicolumn=True, position='H', longtable=True)
with open('Table2.tex', "w") as file:
    file.write(latex_context)